hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -894,25 +894,45 @@ public void merge(Address newTargetAddress) {
             lifecycleService.runUnderLifecycleLock(new Runnable() {
```

Comment:
Can you make this a normal inner class. This method has gotten too long to deal with.